website/src/content/development/data-fetching.mdx

```diff
@@ -0,0 +1,281 @@
+
+The Apollo Client hooks provide an `error` object that we can use in our component to handle failures.
+
+For example, we can render a content notification error instead of rendering the component.
+
+```js title="channels.js" highlightLines="7,13-19"
+import { useQuery } from '@apollo/client/react';
+import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
+import { ContentNotification } from '@commercetools-uikit/notifications';
+import Text from '@commercetools-uikit/text';
+
+const Channels = (props) => {
+  const { data, error } = useQuery(FetchChannelsQuery, {
+    context: {
+      target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
+    }
+  });
+
+  if (error) {
+    return (
+      <ContentNotification type="error">
+        <Text.Body>{getErrorMessage(error)}</Text.Body>
+      </ContentNotification>
+    );
+  }
+
+  return (
+    <div>
+      {/* Do something with `data` */}
+    </div>
+  );
+};
+```
+
+For more information about using the `error` from Apollo Client, see [Error handling](https://www.apollographql.com/docs/react/data/error-handling/).
+
+In our case we simply want to print out the `error.graphQLErrors`. We can attempt to do that by implementing a helper function like this:
+
+```js title="channels.js"
+const getErrorMessage = (error) =>
+  error.graphQLErrors?.map((e) => e.message).join('\n') || error.message;
+```
+
+# Advanced usage
+
+## Using Connector Hooks
+
+As your data requirements grow, for example having multiple queries and mutations in a React component, or using the same query in multiple components, you might want to extract your queries into separate re-usable hooks.
+
+We recommend to extract the logic into what we call "connector hooks."
+
+A connector hook is a React hook that implements most of the data requirements (queries and mutations) specific to the use cases where the data should be used.
+
+For example, we can extract fetching of Channels into a connector hook named `useChannelsFetcher`.
+This hook can be put into a file named `use-channels-connector.js`.
+
+```js title="use-channels-connector.js"
+import { useQuery } from '@apollo/client/react';
+import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
+
+export const useChannelsFetcher = () => {
+  const { data, error, loading } = useQuery(FetchChannelsQuery, {
+    context: {
+      target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
+    }
+  });
+
+  return {
+    channels: data?.channels,
+    error,
+    loading,
+  };
+};
+```
+
+<Info>
+
+Note that the `use-channels-connector.js` file can contain multiple connector hooks. For example `useChannelsFetcher`, `useChannelsUpdater`, `useChannelsCreator`, and so on.
+
+</Info>
+
+We would then use our connector hook in the `Channels` component instead of using the Apollo Client hooks directly. Therefore, there is less code in the React component as most of the logic and configuration is abstracted away in the connector hook.
```

Comment:
```suggestion
We would then use our connector hook in the `Channels` component instead of directly using the Apollo Client hooks. Therefore, there is less code in the React component as most of the logic and configuration is abstracted away in the connector hook.
```